test/Unit/Task/Git/CommitMessageTest.php

```diff
@@ -206,6 +219,32 @@ function () {
                 'Line 5 of commit message has > 10 characters.'
             )
         ];
+        yield 'enforce_text_with_regular_gives_back_message' => [
```

Comment:
What about this instead?

It introduces a new method `buildFailureMessage` which takes the original commit message and the lines it should match on? This keeps te tests readable and gives feedback on the intention.

```php
        yield 'enforce_capitalized_subject' => [
            [
                'enforce_single_lined_subject' => false,
                'max_subject_width' => 10,
                'max_body_width' => 10,
            ],
            $this->mockCommitMsgContext($commitMsg = $this->buildMessage(
                'Subject 1234567891011',
                'Body 1234567891011',
                'Body ok',
                'Body 1110987654321'
            )),
            function () {
            },
            $this->buildFailureMessage(
                ['Please keep the subject <= 10 characters.',
                'Line 3 of commit message has > 10 characters.',
                'Line 5 of commit message has > 10 characters.',],
                $commitMsg,
            )
        ];
```

I wouldn't introduce 2 tests covering the capitalized subject just to check the original commit message. We want to test the logic and validate if the error message contains what what went wrong + the original commit message. All in 1 test-case

Reply:
As all the cases are already covered, I added two new test cases that overlap with some existing tests (that's why 2 tests cover the capitalized subject and 2 tests cover the subject/body width). We can delete the first one as you suggested, no problem.

Also, what should be the content of `buildFailureMessage`? Exactly the same as `withCommitMessage` that we defined above, doesn't it ? Doing so, the only thing we're testing is that the function is called. That's why I tested against a full string.

Reply:
Yes, it would have similar contents to the `withCommitMessage`.
It would validate that the error message contains the expected original commit message.
This needs to be done at alle places where it is added, otherwise one could remove the original commit message without the tests resulting in failure.

Reply:
Added